src/mantle/data/geojson.ts

```diff
@@ -10,62 +10,82 @@ export async function fetchGeoJSON(
   options?: FetchOptions,
 ): Promise<Feature[] | void> {
   const d = data.url ? await (await f(data.url, options)).json() : data.value;
-  return processGeoJSON(d, range);
+  return processGeoJSON(d, range, options?.isSketchLayer);
 }
 
-export function processGeoJSON(geojson: GeoJSON, range?: DataRange): Feature[] {
+export function processGeoJSON(
+  geojson: GeoJSON,
+  range?: DataRange,
+  usePropertyId?: boolean,
+): Feature[] {
   if (geojson.type === "FeatureCollection") {
-    return geojson.features.flatMap(f => processGeoJSON(f, range));
+    return geojson.features.flatMap(f => processGeoJSON(f, range, usePropertyId));
   }
 
   if (geojson.type === "Feature") {
     const geo = geojson.geometry;
     if (geo.type === "GeometryCollection") {
       return geo.geometries.flatMap(geometry => {
-        return processGeoJSON({
-          ...geojson,
-          geometry,
-        });
+        return processGeoJSON(
+          {
+            ...geojson,
+            geometry,
+          },
+          undefined,
+          usePropertyId,
+        );
       });
     }
     if (geo.type === "MultiPoint") {
       return geo.coordinates.flatMap(coord => {
-        return processGeoJSON({
-          ...geojson,
-          geometry: {
-            type: "Point",
-            coordinates: coord,
+        return processGeoJSON(
+          {
+            ...geojson,
+            geometry: {
+              type: "Point",
+              coordinates: coord,
+            },
           },
-        });
+          undefined,
+          usePropertyId,
+        );
       });
     }
     if (geo.type === "MultiLineString") {
       return geo.coordinates.flatMap(coord => {
-        return processGeoJSON({
-          ...geojson,
-          geometry: {
-            type: "LineString",
-            coordinates: coord,
+        return processGeoJSON(
+          {
+            ...geojson,
+            geometry: {
+              type: "LineString",
+              coordinates: coord,
+            },
           },
-        });
+          undefined,
+          usePropertyId,
+        );
       });
     }
     if (geo.type === "MultiPolygon") {
       return geo.coordinates.flatMap(coord => {
-        return processGeoJSON({
-          ...geojson,
-          geometry: {
-            type: "Polygon",
-            coordinates: coord,
+        return processGeoJSON(
+          {
+            ...geojson,
+            geometry: {
+              type: "Polygon",
+              coordinates: coord,
+            },
           },
-        });
+          undefined,
+          usePropertyId,
+        );
       });
     }
 
     return [
       {
         type: "feature",
-        id: generateRandomString(12),
+        id: usePropertyId ? geojson?.properties?.id : generateRandomString(12),
```

Comment:
We have [`idProperty` in Data type](https://github.com/reearth/core/blob/acd9e2b756a6405fccbda71d4e5fdab19f1779bb/src/mantle/types/index.ts#L78), so I think you can use this. How do you think?

Reply:
Ahh i forgot this, i can try this.

Reply:
Updated. Now it's checking `idProperty` instead.